sdk/core/azure-core/src/http/curl/curl.cpp

```diff
@@ -1358,10 +1358,12 @@ std::unique_ptr<CurlNetworkConnection> CurlConnectionPool::ExtractOrCreateCurlCo
   }
 
   long sslOption = 0;
+#if !(LIBCURL_VERSION_MAJOR <= 7 && LIBCURL_VERSION_MINOR < 44)
```

Comment:
nit: I am trying to understand the correctness of this condition again, and I think there is one edge case where it is incorrect, and that is anything `<= 7` but with a `minor version >= 44`, like 6.44. Such a version doesn't exist today, and it is very unlikely that libcurl would introduce >= `.44` versions, on older major versions, but wanted to call it out in case we want to fix it.

We want to enter the code block for any version >= 7.44, correct, and not enter this code path for any lower version?

Let's imagine these test cases and the expected/actual output of the entire `#if` expression:

Libcurl version | Expected | Actual
-- | -- | --
6.1 | FALSE | FALSE
6.51 | FALSE | TRUE
7.1 | FALSE | FALSE
7.51 | TRUE | TRUE
8.1 | TRUE | TRUE
8.51 | TRUE | TRUE

Shall we fix that?

Reply:
This is correct, I should have just gone with @Jinming-Hu proposal first time around 😭  I have fixed this now